packages/dds/tree/src/test/shared-tree/schematizingTreeView.spec.ts

```diff
@@ -100,54 +99,68 @@ describe("SchematizingSimpleTreeView", () => {
-					});
-
-					const root = new Root({ content: 5 });
-
-					const inner = getKernel(root).getOrCreateInnerNode();
-					const field = inner.getBoxed(brand("content"));
-					const child = field.boxedAt(0) ?? assert.fail("Expected child");
-					assert(child instanceof UnhydratedFlexTreeNode);
-
-					// Modify the tree so that it is out of schema.
-					// The public API is supposed to prevent out of schema trees,
-					// so this hack using internal APIs is needed a workaround to test the additional schema validation layer.
-					// In production cases this extra validation exists to help prevent corruption when bugs
-					// allow invalid data through the public API.
-					(child.data as Mutable<typeof child.data>).value = "invalid value";
-
-					// Attempt to initialize with invalid content
-					if (enableSchemaValidation || additionalAsserts) {
-						assert.throws(
-							() => view.initialize(root),
-							validateUsageError(/Tree does not conform to schema./),
-						);
+		it("tolerates staged allowed types", () => {
+			const schemaFactoryAlpha = new SchemaFactoryAlpha("shared tree tests");
+			const stagedSchema = schemaFactoryAlpha.arrayAlpha("TestArray", [
+				SchemaFactoryAlpha.number,
+				SchemaFactoryAlpha.staged(SchemaFactoryAlpha.string),
+			]);
 
-						assert.throws(
-							() => view.root,
-							validateUsageError(/invalid state by another error/),
-						);
-					} else {
-						view.initialize(root);
-						assert.equal(view.root.content, "invalid value");
-					}
-				});
-			}
-		}
+			const emptyContent = {
+				schema: emptySchema,
+				initialTree: undefined,
+			};
+			const checkout = checkoutWithContent(emptyContent);
+			const view = new SchematizingSimpleTreeView(
+				checkout,
+				new TreeViewConfiguration({ schema: stagedSchema }),
+				new MockNodeIdentifierManager(),
+			);
+
+			const { compatibility } = view;
+			assert.equal(compatibility.canView, false);
+			assert.equal(compatibility.canUpgrade, false);
+			assert.equal(compatibility.canInitialize, true);
+
+			view.initialize([5, "test"]);
+			assert.equal(view.root[1], "test");
+
+			assert.throws(
+				() => view.initialize([5, "test"]),
+				validateUsageError(/initialized more than once/),
+			);
+		});
+
+		it("initialize doesn't run schema validation", () => {
+			class Root extends schema.object("Root", {
+				content: SchemaFactoryAlpha.number,
+			}) {}
+
+			const config2 = new TreeViewConfiguration({
+				schema: Root,
+			});
+
+			const view = getView(config2, {
+				forest: ForestTypeReference,
+			});
+
+			const root = new Root({ content: 5 });
+
+			const inner = getKernel(root).getOrCreateInnerNode();
+			const field = inner.getBoxed(brand("content"));
+			const child = field.boxedAt(0) ?? assert.fail("Expected child");
+			assert(child instanceof UnhydratedFlexTreeNode);
+
+			// Modify the tree so that it is out of schema.
+			// The public API is supposed to prevent out of schema trees,
```

Comment:
this is no longer true: staged allowed types allow the public APi to create out of schema trees! Therefor we must absolutely detect this with validation. The lack of validation (no exception) shown below is a blocker for this change as it allows normal looking code using this feature to corrupt documents.